build.gradle.kts

```diff
@@ -0,0 +1,277 @@
+import jetbrains.sign.GpgSignSignatoryProvider
+import org.gradle.jvm.tasks.Jar
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import plugins.publishing.*
+
+/*
+ * Copyright 2000-2021 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+buildscript {
+    repositories {
+        maven { url = uri("https://packages.jetbrains.team/maven/p/jcs/maven") }
+    }
+    dependencies {
+        classpath("com.jetbrains:jet-sign:38")
+    }
+}
+
+repositories {
+    mavenCentral()
+}
+
+plugins {
+    kotlin("multiplatform") version "1.9.22"
+    id("io.github.gradle-nexus.publish-plugin") version "1.1.0"
+    `maven-publish`
+    signing
+    java
+}
+
+var projectVersion = project.findProperty("projectVersion") as String
+val publishingUser: String? = System.getenv("PUBLISHING_USER")
+val publishingPassword: String? = System.getenv("PUBLISHING_PASSWORD")
+if (publishingPassword == null) {
+    projectVersion += "-SNAPSHOT"
+}
+println("##teamcity[setParameter name='java.annotations.version' value='$projectVersion']")
+
+// https://github.com/gradle/gradle/issues/847
+group = "org.jetbrains.proto"
+version = projectVersion
+
+kotlin {
+    androidNativeArm32()
+    androidNativeArm64()
+    androidNativeX86()
+    androidNativeX64()
+
+    iosArm64()
+    iosX64()
+    iosSimulatorArm64()
+
+    js(IR) {
+        browser {}
+    }
+
+    jvmToolchain(8)
+    jvm {
+        compilations.all {
+            kotlinOptions.jvmTarget = "1.8"
+        }
+        withJava()
+    }
+
+    linuxArm64()
+    linuxX64()
+
+    macosX64()
+    macosArm64()
+
+    mingwX64()
+
+    tvosArm64()
+    tvosX64()
+    tvosSimulatorArm64()
+
+    @Suppress("OPT_IN_USAGE")
+    wasmJs {
+        browser()
+    }
+    @Suppress("OPT_IN_USAGE")
+    wasmWasi {
+        nodejs()
+    }
+
+    watchosArm32()
+    watchosArm64()
+    watchosDeviceArm64()
+    watchosX64()
+    watchosSimulatorArm64()
+
+    sourceSets {
+        val commonMain by getting {
+            dependencies {
+                compileOnly("org.jetbrains.kotlin:kotlin-stdlib")
+            }
+        }
+
+        val nonJvmMain by creating {
+            dependsOn(commonMain)
```

Comment:
Since we have a non-standard source set layout here, there's a warning reported by Kotlin Gradle plugin:
```
The Default Kotlin Hierarchy Template was not applied to 'root project 'annotations'':
Explicit .dependsOn() edges were configured for the following source sets:
....
```
You can set the gradle property kotlin.mpp.applyDefaultHierarchyTemplate=false to suppress it.